### PR TITLE
Fix errors in managed resources reconciliation

### PR DESCRIPTION
**Pull request by jarrpa**
This PR fixes a few issues found when trying to reconcile the auxiliary managed resources.

Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

### CODE REVIEW

**Review by umangachapagain — COMMENTED** [review]
lgtm. waiting for tests.

**Comment by jarrpa** [comment]
I have manually verified that StorageClasses are updated appropriately now. The other resources also respond to toggling the reconcile strategy.

**Review by nbalacha — COMMENTED** [review]
looks good

File: pkg/controller/storagecluster/storageclasses.go [hunk 1]
```diff
@@ -66,8 +66,6 @@ func (r *ReconcileStorageCluster) createStorageClasses(scs []*storagev1.StorageC
 					return err
 				}
 				reqLogger.Info(fmt.Sprintf("Creating StorageClass %s", sc.Name))
-				existing.ObjectMeta.OwnerReferences = sc.ObjectMeta.OwnerReferences
-				sc.ObjectMeta = existing.ObjectMeta
```
Comment (iamniting) [inline pkg/controller/storagecluster/storageclasses.go:70]: Don't we want to keep anything from the existing storage class like labels and annotations etc.

Comment (jarrpa) [reply pkg/controller/storagecluster/storageclasses.go:70]: No. We are reconciling, we own the StorageClass, that means that any modifications to the StorageClass must be overridden.

**Review by obnoxxx — APPROVED** [review]
/lgtm

**Review by agarwal-mudit — APPROVED** [review]
/lgtm

**Pull request merged**